### ISSUE TITLE
Make task scheduler work in windows when computer shut down? 

### ISSUE THREAD

**Issue by aliifam** [JareBear12418/Daily-Git-Commit#2]
What solution if I want the program still running when my laptop is shut down?

**Comment by Jared-Gross** [comment]
If your laptop is shut off, then I don't believe you can run anything at anytime.

However if your laptops lid is closed it should still run **IF** you change the settings for when your lid is closed to **NOT** enter sleep mode. `Control Panel\Hardware and Sound\Power Options\System Settings` Here Make sure your laptop never goes off (goes into sleep mode) if you want this to work.

You can't run programs if the computer is not running, they run on a schedule/idle/login/logoff/startup/event...etc. 

Effectively how I use it, is my laptop NEVER sleeps, NEVER turns off, I just turn off the screen, not the computer itself. Then I can run my program on a daily schedule or how ever I please, I have it every couple minutes. This works because the computer **ISNT** turned off. 

In short, you **CANT** run a program, while the laptop is shut down. You can give the illusion its shut down, by just turning off the screen instead.

**Comment by aliifam** [comment]
> If your laptop is shut off, then I don't believe you can run anything at anytime.
> 
> However if your laptops lid is closed it should still run **IF** you change the settings for when your lid is closed to **NOT** enter sleep mode. `Control Panel\Hardware and Sound\Power Options\System Settings` Here Make sure your laptop never goes off (goes into sleep mode) if you want this to work.
> 
> You can't run programs if the computer is not running, they run on a schedule/idle/login/logoff/startup/event...etc.
> 
> Effectively how I use it, is my laptop NEVER sleeps, NEVER turns off, I just turn off the screen, not the computer itself. Then I can run my program on a daily schedule or how ever I please, I have it every couple minutes. This works because the computer **ISNT** turned off.
> 
> In short, you **CANT** run a program, while the laptop is shut down. You can give the illusion its shut down, by just turning off the screen instead.

Can this script is working online, I mean the script is work in github action or other services?

**Comment by Jared-Gross** [comment]
I don't know how you would do that, you'd have to figure it out yourself. Alternatively you could simply use a raspberry pi too, or a server of yours.

To answer your question, I don't know.

**Comment by aliifam** [comment]
> I don't know how you would do that, you'd have to figure it out yourself. Alternatively you could simply use a raspberry pi too, or a server of yours.
> 
> To answer your question, I don't know.

how to run this program in virtual environment?

**Comment by Jared-Gross** [comment]
Have a look here as to how to set up a [Virtual environment](https://docs.python.org/3/library/venv.html) 

[Here's another good source](https://realpython.com/lessons/creating-virtual-environment/)
[Here's one more](https://www.javatpoint.com/how-to-create-a-virtual-environment-in-python)

There's plenty of information online to find out for yourself.

Make sure you **activate** the virtual environment before you attempt to install the requirements. 

To install the requirements. You can install like this: `pip install -r requirements.txt` or like this: `pip install gitpython pyyaml`
